src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java

```diff
@@ -0,0 +1,378 @@
+    Keycloak keycloak = getAdminClient();
+    RealmResource realm = keycloak.realm(realmName);
+    UsersResource users = realm.users();
+    ClientRepresentation client = getClient(keycloak);
+    
+    if (client == null) {
+      throw new AuthorizationException(AUTHORIZATION_EXCEPTION_MESSAGE);
+    }
+    
+    UserPoliciesResource userPolicies = realm.clients().get(client.getId()).authorization().policies().user();
+    List<UserRepresentation> existingUsers = users.list().stream().collect(Collectors.toList());
+    
+    for (UserRepresentation existingUser : existingUsers) {
+      UUID userId = UUID.fromString(existingUser.getId());
+      String userName = existingUser.getUsername();
+      String policyNameTemplate = String.format("user-%s", userName);
+      result.add(userId);
+      UserPolicyRepresentation policyRepresentation = userPolicies.findByName(policyNameTemplate);
+     
+      if (policyRepresentation == null) {
+
+        policyRepresentation = new UserPolicyRepresentation();
+        policyRepresentation.setName(policyNameTemplate);
+        policyRepresentation.setDecisionStrategy(DecisionStrategy.AFFIRMATIVE);
+        policyRepresentation.addUser(userName);
+        policyRepresentation.setId(userId.toString());
+        userPolicies.create(policyRepresentation);
+      }
+    }
+    
+     return result; 
+  }
+  
+  /**
+   * Finds a id from Keycloak create response 
+   * 
+   * @param response response object
+   * @return UUID create response id
+   */
+  public UUID getCreateResponseId(Response response) {
+    if (response.getStatus() != 201) {
+      try {
+        if (logger.isErrorEnabled()) {
+          logger.error("Failed to execute create: {}", IOUtils.toString((InputStream) response.getEntity(), StandardCharsets.UTF_8));
+        }
+      } catch (IOException e) {
+        logger.error("Failed to extract error message", e);
+      }
+      
+      return null;
+    }
+   
+    String location = response.getHeaderString("location");
+    if (StringUtils.isBlank(location)) {
+      return null;
+    }
+    
+    Pattern pattern = Pattern.compile(".*\\/(.*)$");
+    Matcher matcher = pattern.matcher(location);
+    
+    if (matcher.find()) {
+      return UUID.fromString(matcher.group(1));
+    }
+    
+    return null;
+  }
+  
+  /**
+   * Evaluates eligibility of a user to access resources
+   * 
+   * @param loggedUserId UUID user id
+   * @param scopes Authorization scopes
+   * @return list of UUID id's that resources refer to
+   */
+  public List<UUID> resourceAccessEvaluate(UUID loggedUserId) {
+    try {
+      List<AuthorizationScope> scopes = Collections.singletonList(AuthorizationScope.STORY_ACCESS);
+      Keycloak keycloak = getAdminClient();
+      RealmResource realm = keycloak.realm(REALM);
+      ClientRepresentation client = getClient(keycloak);
+      List<UUID> storiesIdList = storyController.listStories().stream().map(Story::getId).collect(Collectors.toList());
```

Comment:
Use list of stories from method parameter instead of loading them in this method (see comment above)